style_editor.html

```diff
@@ -9280,37 +9490,220 @@
+  } else {
+    darkButton.checked = false;
+    darkOn = false;
+    body.classList.remove("dark-mode");
+    structuredView.classList.remove("dark-mode");
+  }
+  // console.log("BC-darkOn = " + darkOn);
+
+  // TIPS_BUTTON
+  var tipsButton = FIND("TIPS_BUTTON");
+  var tipsState = getSavedCookie("tipsState");
+  // console.log("BC-getSavedCookie called, tipsState = " + tipsState);
+  if (tipsState !== "false") {
+    tipsButton.checked = true;
+    tipsOn = true;
+    body.classList.remove("tooltips-disabled");
+    const elementsWithDataTitles = document.querySelectorAll("[data-title]");
+    elementsWithDataTitles.forEach((element) => {
+      element.setAttribute("title", element.dataset.title);
+      element.removeAttribute("data-title");
+    });
+  } else {
+    tipsButton.checked = false;
+    tipsOn = false;
+    body.classList.add("tooltips-disabled");
+    const elementsWithTitles = document.querySelectorAll("[title]");
+    elementsWithTitles.forEach((element) => {
+      element.dataset.title = element.getAttribute("title");
+      element.removeAttribute("title");
+    });
+  }
+  // console.log("BC-tipsOn = " + tipsOn);
+
+  // STILL_BUTTON
+  var stillButton = FIND("STILL_BUTTON");
+  var stillState = getSavedCookie("stillState");
+  console.log("BC-getSavedCookie called, stillState = " + stillState);
+  if (stillState === "true") {
+    stillButton.checked = true;
+    stillOn = true;
+  }
+  // console.log("BC-stillOn = " + stillOn);
+
+
+  // SWING_EMULATION_BUTTON
+  var swingEmButton = FIND("SWING_EMULATION_BUTTON");
+  var swingEmState = getSavedCookie("swingEmState");
+  // console.log("BC-getSavedCookie called, swingEmState = " + swingEmState);
+  if (swingEmState === "true") {
+    swingEmButton.checked = true;
+    swingEmOn = true;
+  }
+  // console.log("BC-swingEmOn = " + swingEmOn);
+
+  // INHILT_BUTTON
+  var inhiltButton = FIND("INHILT_BUTTON");
+  var inhiltState = getSavedCookie("inhiltState");
+  // console.log("BC-getSavedCookie called, inhiltState = " + inhiltState);
+  if (inhiltState === "true") {
+    inhiltButton.checked = true;
+    inhiltOn = true;
+    STATE_NUM_LEDS = 145 - STATE_NUM_LEDS;
+  }
+  // console.log("BC-inhiltOn = " + inhiltOn);
+
+  // SLOW_BUTTON
+  var slowButton = FIND("SLOW_BUTTON");
+  var slowState = getSavedCookie("slowState");
+  // console.log("BC-getSavedCookie called, slowState = " + slowState);
+  if (slowState === "true") {
+    slowButton.checked = true;
+    slowOn = true;
+    framesPerUpdate = 60 - framesPerUpdate;
+  }
+  // console.log("BC-slowOn = " + slowOn);
+
+  // WAVLEN_VALUE
+  var wavlenInput = FIND("WAVLEN_VALUE");
+  wavlenValue = parseInt(getSavedCookie("wavlenValue"));
+  if (!wavlenValue) {
+    wavlenValue = 500;
+  }
+  // console.log("BC-getSavedCookie called, wavlenValue = " + wavlenValue);
+  wavlenInput.value = wavlenValue;
+  myWavLen.setLength(wavlenValue);
+wavlenInput.addEventListener("click", function(e) {
```

Comment:
This may look like a lot of listeners, but they make the value entry better.
Single clicking in the field selects all digits, so just starting typing starts a total new entry as opposed to requiring clicking an insertion point and needing to backspace or delete.
The input value is safeguarded against leaving an empty value, which can cause issues.
If all digits deleted, a minimum value of 1 is autofilled, and left selected so that entry of a different value is easier.
Hitting enter "enters" the value, just removing focus from the box.

Reply:
I don't think we should do this.
Does it work on mobile?
Does it work on all browsers?
Does it work with screen readers?
Making things better is hard, it's often best to just leave them the way they are.

Reply:
Yes.
Yes.
I don't have a screen reader.... can you build blade style's on that?
It's a safeguard in place. Example: If a user sticks WavLen as an AlphaL function, having no value creates a repeating loop of "get color" errors that will likely anger the browser.
It is working great, I was just saying I know it looks like a lot for a little value entry, but I threw the kitchen sink at it to make sure it's good.